src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+// same as func_80042244, but tests if WaterBox properties & 0x80000 != 0
+// returns true if point is within the xz boundaries of an active water box, else false
+// `ySurface` returns the water box's surface, while `outWaterBox` returns a pointer to the WaterBox
+s32 func_800425B0(GlobalContext* globalCtx, CollisionContext* colCtx, f32 x, f32 z, f32* ySurface,
+                  WaterBox** outWaterBox) {
+    CollisionHeader* colHeader;
+    u32 room;
+    WaterBox* curWaterBox;
+
+    colHeader = colCtx->colHeader;
+    if (colHeader->nbWaterBoxes == 0 || colHeader->waterBoxes == PHYSICAL_TO_VIRTUAL(gSegments[0])) {
+        return false;
+    }
+
+    for (curWaterBox = colHeader->waterBoxes; curWaterBox < colHeader->waterBoxes + colHeader->nbWaterBoxes;
+         curWaterBox++) {
+        room = (curWaterBox->properties >> 0xD) & 0x3F;
+        if ((room == globalCtx->roomCtx.curRoom.num) || (room == 0x3F)) {
+            if ((curWaterBox->properties & 0x80000) != 0) {
+                if (curWaterBox->xMin < x && x < (curWaterBox->xMin + curWaterBox->xLength)) {
+                    if (curWaterBox->zMin < z && z < (curWaterBox->zMin + curWaterBox->zLength)) {
+                        *outWaterBox = curWaterBox;
+                        *ySurface = curWaterBox->ySurface;
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+    return false;
+}
+
+// Gets the `closestPoint` to `point` on the line formed from the intesection of planes `polyA` and `polyB`
+// returns true if the `closestPoint` exists, else returns false
+s32 func_80042708(CollisionPoly* polyA, CollisionPoly* polyB, Vec3f* point, Vec3f* closestPoint) {
+    f32 n1X;
+    f32 n1Y;
+    f32 n1Z;
+    f32 n2X;
+    f32 n2Y;
+    f32 n2Z;
+
+    CollisionPoly_GetNormal(polyA, &n1X, &n1Y, &n1Z);
+    CollisionPoly_GetNormal(polyB, &n2X, &n2Y, &n2Z);
+    return Math3D_PlaneVsPlaneVsLineClosestPoint(n1X, n1Y, n1Z, polyA->dist, n2X, n2Y, n2Z, polyB->dist, point,
+                                                 closestPoint);
+}
+
+// Get the `closestPoint` to line (`pointA`, `pointB`) formed from the intersection of planes `polyA` and `polyB`
+// returns true if the `closestPoint` exists, else returns false
+s32 func_800427B4(CollisionPoly* polyA, CollisionPoly* polyB, Vec3f* pointA, Vec3f* pointB, Vec3f* closestPoint) {
+    f32 n1X;
+    f32 n1Y;
+    f32 n1Z;
+    f32 n2X;
+    f32 n2Y;
+    f32 n2Z;
+    s32 result;
+
+    CollisionPoly_GetNormal(polyA, &n1X, &n1Y, &n1Z);
+    CollisionPoly_GetNormal(polyB, &n2X, &n2Y, &n2Z);
+    result = Math3D_PlaneVsLineSegClosestPoint(n1X, n1Y, n1Z, polyA->dist, n2X, n2Y, n2Z, polyB->dist, pointA, pointB,
+                                               closestPoint);
+    return result;
+}
+
+// Draw a list of dyna polys, specified by `nodeId`
+void func_80042868(GlobalContext* globalCtx, CollisionContext* colCtx, DynaCollisionContext* dyna, u16* nodeId, u8 r,
```

Comment:
naming the draw functions from here to the bottom of the file should be doable too i think